pxf/pxf-ignite/src/main/java/org/apache/hawq/pxf/plugins/ignite/IgnitePartitionFragmenter.java

```diff
@@ -0,0 +1,411 @@
+                    byte[] msEnd = ByteUtil.getBytes(fragEnd.getTimeInMillis());
+                    fragmentMetadata = ArrayUtils.addAll(msStart, msEnd);
+
+                    Fragment fragment = new Fragment(inputData.getDataSource(), replicaHostAddressWrapped, fragmentMetadata, fragmentUserdata);
+                    fragments.add(fragment);
+
+                    // Continue the previous fragment
+                    fragStart = fragEnd;
+                }
+                break;
+            }
+            case INT: {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("getFragments() found INT partition");
+                }
+                int rangeStart = Integer.parseInt(range[0]);
+                int rangeEnd = Integer.parseInt(range[1]);
+                int currInterval = intervalNum;
+
+                int fragStart = rangeStart;
+                while (fragStart < rangeEnd) {
+                    int fragEnd = fragStart + currInterval;
+                    if (fragEnd > rangeEnd) {
+                        fragEnd = rangeEnd;
+                    }
+
+                    byte[] bStart = ByteUtil.getBytes(fragStart);
+                    byte[] bEnd = ByteUtil.getBytes(fragEnd);
+                    fragmentMetadata = ArrayUtils.addAll(bStart, bEnd);
+
+                    Fragment fragment = new Fragment(inputData.getDataSource(), replicaHostAddressWrapped, fragmentMetadata, fragmentUserdata);
+                    fragments.add(fragment);
+
+                    // Continue the previous fragment
+                    fragStart = fragEnd;
+                }
+                break;
+            }
+            case ENUM: {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("getFragments() found ENUM partition");
+                }
+                for (String frag : range) {
+                    fragmentMetadata = frag.getBytes();
+                    Fragment fragment = new Fragment(inputData.getDataSource(), replicaHostAddressWrapped, fragmentMetadata, fragmentUserdata);
+                    fragments.add(fragment);
+                }
+                break;
+            }
+            default: {
+                throw new UnsupportedOperationException("getFragments() found a partition of unknown type and failed");
+            }
+        }
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("getFragments() successful");
+        }
+        return fragments;
+    }
+
+    /**
+     * Insert partition constraints into the prepared SQL query.
+     * 
+     * @param inputData pre-validated PXF InputData
+     * @param sb the SQL query that is prepared for appending WHERE constraints.
+     * Other SQL statements may be present, but they must be complete. Note that no check is performed to check their "completeness"
+     */
+    public static StringBuilder buildFragmenterSql(InputData inputData, StringBuilder sb) {

```

Comment:
Do we really want to mutate input parameter and then return it?

Reply:
The `StringBuilder sb` may include other WHERE constraints (e.g., partition constraints), thus the result of `buildFragmenterSql()` strongly depends on the contents of `sb`. 

Consider the following example. Let `buildFragmenterSql` return a `String`. In some cases the return value will be like `WHERE id=10 AND ...`, while in other cases ` AND id=10 AND ...`. In the second case, the output has no sense without the initial query, so both the initial query and the result of the function must be present.

Besides, passing a mutable parameter here points to the fact that the existing `WHERE` constraints (present in `StringBuilder sb`) must be complete and that the query will be changed accordingly.

Reply:
Since you are mutating initial query - I think it would be cleaner to do it in place and return `void`.

Reply:
Thank you! This is a better solution. Done.